publish-pipeline/publish-core/src/test/scala/org/sunbird/job/publish/spec/ThumbnailGeneratorSpec.scala

```diff
@@ -37,6 +37,19 @@ class ThumbnailGeneratorSpec extends FlatSpec with BeforeAndAfterAll with Matche
     resultMetadata.getOrElse("appIcon", "").asInstanceOf[String].isEmpty should be(false)
   }
 
+  "Object Thumbnail Generator generateThumbnail with google drive link" should "add the thumbnail to ObjectData" in {
+    val hierarchy = Map("identifier" -> "do_123", "children" -> List(Map("identifier" -> "do_234", "name" -> "Children-1"), Map("identifier" -> "do_345", "name" -> "Children-2")))
+    val metadata = Map("identifier" -> "do_123", "appIcon" -> "https://drive.google.com/uc?export=download&id=1-dFzAeSNmx1ZRn77CEntyQA-VcBE0PKg", "IL_UNIQUE_ID" -> "do_123", "objectType" -> "QuestionSet", "name" -> "Test QuestionSet", "status" -> "Live")
+    val objData = new ObjectData("do_123", metadata, None, Some(hierarchy))
+    val thumbnailGenerator = new TestThumbnailGenerator()
+    val obj = thumbnailGenerator.generateThumbnail(objData)
+    val result = obj.getOrElse(objData)
+    val resultMetadata = result.metadata
+    resultMetadata.isEmpty should be(false)
+    resultMetadata.getOrElse("posterImage", "").asInstanceOf[String].isEmpty should be(false)
+    resultMetadata.getOrElse("appIcon", "").asInstanceOf[String].isEmpty should be(false)
```

Comment:
```suggestion
    resultMetadata.getOrElse("appIcon", "").asInstanceOf[String].isEmpty should be(false)
```
Can you please assert the value of appIcon?

Reply:
done